File: app/models/proyecto_persona.py
from datetime import date

from sqlalchemy import ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class ProyectoPersona(Base):
    __tablename__ = "proyecto_persona"

    id: Mapped[int] = mapped_column(primary_key=True)
    id_proyecto: Mapped[str] = mapped_column(ForeignKey("proyecto.id_proyecto"))
    persona_id: Mapped[int] = mapped_column(ForeignKey("persona.id_persona"))
    id_rol: Mapped[int] = mapped_column(ForeignKey("rol.id_rol"))
    horas_semanales: Mapped[int]
    fecha_inicio: Mapped[date]
    fecha_fin: Mapped[date | None]

File: flutter_app/lib/models/proyecto_persona.dart
class ProyectoPersona {
  final int id;
  final String idProyecto;
  final int personaId;
  final int idRol;
  final int horasSemanales;
  final DateTime fechaInicio;
  final DateTime? fechaFin;

  ProyectoPersona({
    required this.id,
    required this.idProyecto,
    required this.personaId,
    required this.idRol,
    required this.horasSemanales,
    required this.fechaInicio,
    required this.fechaFin,
  });

  factory ProyectoPersona.fromJson(Map<String, dynamic> json) {
    final fechaFin = json['fecha_fin'];
    return ProyectoPersona(
      id: json['id'] as int,
      idProyecto: json['id_proyecto'] as String,
      personaId: json['persona_id'] as int,
      idRol: json['id_rol'] as int,
      horasSemanales: json['horas_semanales'] as int,
      fechaInicio: DateTime.parse(json['fecha_inicio'] as String),
      fechaFin: fechaFin == null ? null : DateTime.parse(fechaFin as String),
    );
  }
}
